provdebug/pvdebug.py

```diff
@@ -92,34 +103,62 @@ def run():
         fst_parsed_file = Parser(provided_files[0])
         snd_parsed_file = Parser(provided_files[1])
         fst_prov_slice = ProvSlice(fst_parsed_file._prov_nodes)
         snd_prov_slice = ProvSlice(snd_parsed_file._prov_nodes)
-        similarity = fst_prov_slice.compute_similarity(snd_prov_slice)
-        if similarity == 1.0:
-            print("The provided provenance data is identical.")
-        elif similarity >= similarity_threshold:
-            divergent_nodes = fst_prov_slice.divergent_nodes(snd_prov_slice)
-            nodes_not_in_fst = divergent_nodes[0]
-            nodes_not_in_snd = divergent_nodes[1]
-            print(f"Nodes not in: {args.diff[0]}")
-            for node in nodes_not_in_fst:
-                node.pretty_print()
-            print("=" * 69)
-            print(f"Nodes not in: {args.diff[1]}")
-            for node in nodes_not_in_snd:
-                node.pretty_print()
-        else:
-            print("The provided provenance data is too dissimilar, manual inspection is encouraged.")
+        while True:
+            global similarity_threshold
+            user_input = input("> ")
+            user_choices = user_input.split(" ")
+            user_flag = user_choices[0]
+            if user_flag == "a" or user_flag == "analyze":
+                similarity = fst_prov_slice.compute_similarity(snd_prov_slice)
+                print(f"Similarity threshold: {similarity_threshold}\nSimilarity score: {similarity}")
+                if similarity == 1.0:
+                    print("The provided provenance data is identical.")
+                elif similarity >= similarity_threshold:
+                    divergent_nodes = fst_prov_slice.divergent_nodes(snd_prov_slice)
+                    nodes_not_in_fst = divergent_nodes[0]
+                    nodes_not_in_snd = divergent_nodes[1]
+                    print(f"Nodes not in: {args.diff[0]}")
+                    for node in nodes_not_in_fst:
+                        node.pretty_print()
+                    print("=" * 69)
+                    print(f"Nodes not in: {args.diff[1]}")
+                    for node in nodes_not_in_snd:
+                        node.pretty_print()
+                else:
+                    print("The provided provenance data is too dissimilar, set a lower similarity threshold with 't' or perform manual inspection.")
+                continue
+            elif user_flag == "h" or user_flag == "help":
+                print(diffHelpText)
+                continue
+            elif user_flag == "t" or user_flag == "threshold":
+                threshold_info_text = '''The similarity threshold is the minimum score required for Diff Mode to proceed analyzing the differences between two scripts. \nDiff Mode computes the similarity score between two scripts using the longest contiguous matching subsequence (LCS).'''
+                print(threshold_info_text)
```

Comment:
Do you have any suggestions for how to elegantly split this string?

Reply:
You can do multi-line string splits by using the `\` symbol in Python